### ISSUE TITLE
[bitnami/mysql] Needs to expose   podSecurityContext/containerSecurityContext for metrics

### ISSUE THREAD

**Issue by udf2457** [bitnami/charts#15552]
### Name and Version

bitnami/mysql 9.6.0

### What architecture are you using?

None

### What steps will reproduce the bug?

Run helm install with restricted PSS enabled.

Note the following warning:

```
W0318 12:58:17.528714    8052 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "metrics" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "metrics" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "metrics" must set securityContext.runAsNonRoot=true)
W0318 12:58:17.528714    8052 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "metrics" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "metrics" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "metrics" must set securityContext.runAsNonRoot=true)
```

Adding:

```yaml
metrics:
  podSecurityContext:
    seccompProfile:
      type: RuntimeDefault
  containerSecurityContext:
    allowPrivilegeEscalation: false
    capabilities:
      drop: [ALL]
```

It does not resolve the issue because the variables are not passed down to metrics.

Please note that I am still a newbie in Helm charts, so please do not ask me to submit a PR!

### Are you using any custom parameters or values?

```yaml
global:
  storageClass: "longhorn"
architecture: "replication"
auth:
  existingSecret: "mysql"
primary:
  extraFlags: "--authentication-policy=mysql_native_password"
  startupProbe:
    initialDelaySeconds: 60
  podSecurityContext:
    seccompProfile:
      type: RuntimeDefault
  containerSecurityContext:
    allowPrivilegeEscalation: false
    capabilities:
      drop: [ALL]
secondary:
  extraFlags: "--authentication-policy=mysql_native_password"
  startupProbe:
    initialDelaySeconds: 60
  podSecurityContext:
    seccompProfile:
      type: RuntimeDefault
  containerSecurityContext:
    allowPrivilegeEscalation: false
    capabilities:
      drop: [ALL]
metrics:
  enabled: true
```

### What is the expected behavior?

No warning.

**Comment by fmulero** [comment]
It seems I was wrong, I have to investigate about this topic a little bit more. 

I've just deploy the chart with the configuration above and I didn't see any issue and the capabilities are set to 0. (for that reason I thought the process couldn't open the ports)

**Comment by udf2457** [comment]
Yes, I originally thought the same as you, but then one day I was deploying something and I forgot to set `NET_BIND_SERVICE` and discovered it worked.

It would be interesting to see what you find on your investigations.  Maybe its an omission or regression in the Kubernetes code base or maybe it works that way becasue of the way the Linux kernel works.  Unfortunatley it's all a bit too low-level and outside of my area of expertise, so I can't offer much help on the detail ....

**Comment by udf2457** [comment]
@fmulero Curiosity got the better of me and I spent far too much time in the company of Mr Google ....

TL;DR:  I _think_ this might be a contributory factor to the answer: https://github.com/moby/moby/pull/41030

**Comment by fmulero** [comment]
I think the reason was in front of us from the begining, in the definition of the [capabilities](https://man7.org/linux/man-pages/man7/capabilities.7.html) we can see this:

> Starting with kernel 2.2, Linux divides the privileges traditionally associated with superuser into distinct units, known as capabilities, which can be independently enabled and disabled. Capabilities are a per-thread attribute.

We are running non privileged containers, so we can drop safely all capabilities because they apply only to privileged processes.
